src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/security/idam/IdamAuthorizor.java

```diff
@@ -81,6 +81,10 @@ private String fetchCodeAuthorization(
                         }
                     ).getBody();
 
+            if (authCode == null) {
+                throw new IllegalStateException("Error in getting auth code from IDAM");
```

Comment:
you could throw RestClientException so try catch block is used and exception is wrapped in IdentityManagerResponseException